packages/graphql_codegen/test/assets/fragments_and_loose_type_requirements/document.graphql.dart

```diff
@@ -71,6 +71,32 @@ extension UtilityExtension$Fragment$F on Fragment$F {
         this,
         (i) => i,
       );
+  _T when<_T>({required _T Function(Fragment$F$$T) t}) {
+    switch ($__typename) {
+      case "T":
+        return t(this as Fragment$F$$T);
+
+      default:
+        throw Exception("Unknown typename '${$__typename}'");
+    }
+  }
+
+  _T maybeWhen<_T>({
+    _T Function(Fragment$F$$T)? t,
+    required _T Function() orElse,
```

Comment:
Following the abovementioned point, this should follow the same decision, e.g., remove the `orElse` and make the return type nullable, or keep it as is.

I am interested in hearing your thoughts on this.

Reply:
settled on using non-null and adding a required `orElse` per top-level conversation